### DISCUSSION TITLE
MLP Support

### DISCUSSION THREAD

**Discussion by tunguz** [intel/scikit-learn-intelex#920] [Ideas]
Hi, I am really impressed with this project. I was just wondering if there are any plans to offer MLP support. I understand that NNs are usually relegated to the NN-specific frameworks (PyTorch, TF/Keras, etc.) but IMHO having an easy and accessible NN also within the skelarn library would be very handy. I have used it for several problems, and it usually gives pretty good results. It is especially useful for ensembling with other skelarn models.

**Comment by Pahandrovich** [comment]
Hi @tunguz,
Thanks for your feedback. Our library is currently specialized in classic machine learning algorithms (no neural networks). You can share which algorithms you use most often. I think this will help us when optimizing some of scikit-learn other algorithms.

**Reply by tunguz** [reply]
Thanks @Pahandrovich for the response. I understand that the "classic" ML algorithms should be a natural priority for a library that aims to accelerate and optimize sklearn. I would just like to have you consider MLP as well - it is usually a **very** good algorithm to ensemble with the rest of the classic logos, and having it included in a single library with the rest of them would make many ML workflows smoother.

**Reply by syakov-intel** [reply]
We have teams working on more specialized [solutions](https://github.com/intel/intel-extension-for-pytorch) .

**Reply by tunguz** [reply]
I am sorry, but that is not the solution that I was hoping to get. A general extension of PyTorch for the Intel platform is fine, but an accelerated sklearn-API for a simple MLP is what I was **specifically** asking for. Thanks.
